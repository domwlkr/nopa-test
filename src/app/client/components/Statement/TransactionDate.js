import React from 'react';

const TransactionDate = (props) => {
  return (
    <span>Hiya</span>
  )
}

export default TransactionDate;