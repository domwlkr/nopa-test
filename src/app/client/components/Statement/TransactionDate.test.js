import React from 'react';
import TransactionDate from './TransactionDate';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

const dateArray = [{'id': 1, 'dateStr': 'Now', 'beneficary': 'PAYPAL ZARA', 'value': '- Â£35.98'}];

test('Renders correctly', () => {
  const transactionDate = renderer.create(
    <TransactionDate date={dateArray} />
  ).toJSON();

  expect(transactionDate).toMatchSnapshot();
});

test('Displays the correct date', () => {
  const transactionDate = shallow(
    <TransactionDate date={dateArray} />
  );

  expect(transactionDate.instance().props.date[0].dateStr).toEqual('Now');
});