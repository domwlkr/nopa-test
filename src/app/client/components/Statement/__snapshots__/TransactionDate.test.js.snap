// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="transaction-date-wrap"
>
  <div
    className="transaction-date"
  >
    Today
  </div>
  <div
    className="transaction"
  >
    <span>
      PAYPAL ZARA
    </span>
    <span>
      - Â£35.98
    </span>
  </div>
</div>
`;
